app/models/concerns/facets/managed_host_extensions.rb

```diff
@@ -0,0 +1,66 @@
+require 'facets'
+
+module Facets
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      Facets::ManagedHostExtensions.refresh_facet_relations(self)
+
+      def attributes
+        hash = super
+
+        # include all facet attributes by default
+        facets_with_definitions.each do |facet, facet_definition|
+          hash["#{facet_definition.name}_attributes"] = facet.attributes.reject { |key, _| %w(created_at updated_at).include? key }
+        end
+        hash
+      end
+    end
+
+    def self.refresh_facet_relations(klass)
+      Facets.registered_facets.values.each do |facet_config|
+        self.register_facet_relation(klass, facet_config)
+      end
+    end
+
+    def self.register_facet_relation(klass, facet_config)
```

Comment:
Please add some docs on this method, like "Defines relation between hosts and facets ...."